File: sumFiles/sumFile.go
package main 

import (
	"os"
	"fmt"
)

func main() {
	
	var a,b int
	
	inFile,err1 := os.Open("adunare.in")
	outFile,err2 := os.Create("adunare.out")
	
	defer func() {
	 if inFile != nil {
	 	 inFile.Close()
	 }
	 
	 if outFile != nil {
	 	outFile.Close()
	 }
		
	 
	 }()
	
	if err1 != nil || err2 != nil {
		panic(err2)
	}
	
	_,err1 = fmt.Fscanf(inFile,"%d \n %d",&a,&b)
	
	fmt.Fprintf(outFile,"%d",a+b) 
}

File: sumFiles/sumFile.c
/*
 * main.c
 *
 *  Created on: Jul 15, 2012
 *      Author: Catalin
 */

#include<stdio.h>


int main(){

	FILE * inFile = fopen("adunare.in","r");
	FILE * outFile = fopen("adunare.out","w");
	int a,b;

	if( inFile == NULL || outFile == NULL ){
		fprintf(stderr,"fopen error");
		return 1;
	}

	fscanf(inFile,"%d \n %d",&a,&b);

	fprintf(outFile,"%d",a+b);

	return 0;

}
